scipy/stats/_multivariate.py

```diff
@@ -2954,6 +2885,59 @@ def var(self, df, scale):
         out = self._var(dim, df, scale)
         return _squeeze_output(out) if out is not None else out
 
+    def _inv_standard_rvs(self, n, shape, dim, df, random_state):
+        """
+        Parameters
+        ----------
+        n : integer
+            Number of variates to generate
+        shape : iterable
+            Shape of the variates to generate
+        dim : int
+            Dimension of the scale matrix
+        df : int
+            Degrees of freedom
+        random_state : {None, int, `numpy.random.Generator`,
+                        `numpy.random.RandomState`}, optional
+
+            If `seed` is None (or `np.random`), the `numpy.random.RandomState`
+            singleton is used.
+            If `seed` is an int, a new ``RandomState`` instance is used,
+            seeded with `seed`.
+            If `seed` is already a ``Generator`` or ``RandomState`` instance
+            then that instance is used.
+
+        Notes
+        -----
+        As this function does no argument checking, it should not be
+        called directly; use 'rvs' instead.
+
+        """
+        # Random normal variates for off-diagonal elements
+        n_tril = dim * (dim-1) // 2
+        covariances = random_state.normal(
+            size=n*n_tril).reshape(shape+(n_tril,))
+
+        # Random chi-square variates for diagonal elements
+        variances = (np.r_[[random_state.chisquare(df+(i+1)-dim, size=n)**0.5
```

Comment:
This method is identical to `_standard_rvs` except for this line, where the diagonal elements are reversed. Might be worth merging the methods.

Reply:
If I am reading this function correctly we are forming nD stacked triangular arrays with a very convoluted way of doing it. 

Especially this line is creating separate arrays putting into a list then unpacking and writing to a bigger array. Instead it can write to zero array diagonal directly at each spin of chisquare without any of these intermediate variables.

Then again, I am not sure what A should be. Might be a good addition if docstring also has a Returns section. There are easier ways to layout the triangular offdiag entries. Probably we should write a low-level function to do this or [`?trttp`](https://netlib.org/lapack/explore-html/da/dba/group__double_o_t_h_e_rcomputational_ga51c36532bd647e5012533f5711786da4.html#ga51c36532bd647e5012533f5711786da4) variants.

Reply:
Here's a much more straightforward approach for filling `A`. Do you see any problems with this?

```python
# Random normal variates for off-diagonal elements
tri_rows, tri_cols = np.tril_indices(dim, k=-1)
n_tril = dim * (dim-1) // 2
A[..., tri_rows, tri_cols] = random_state.normal(size=(*shape, n_tril))

# Random chi variates for diagonal elements
rows = np.arange(dim)
chi_dfs = (df - dim + 1) + rows
A[..., rows, rows] = random_state.chisquare(df=chi_dfs, size=(*shape, dim))**0.5
```

Reply:
I adopted this approach.

Reply:
I don't think this part of the code is performance driven hence much cleaner for me, thanks.